content/post/performance-metrics-overview.md

```diff
@@ -0,0 +1,111 @@
++++
+title = "Building low-overhead metrics collection for high-performance systems"
+date = 2018-02-20T07:20:00-05:00
+draft = false
+slug = "building-low-overhead-metrics-collection-for-high-performance-systems"
+author = "jonbrwn"
+description = "Design decisions that allowed us to capture metrics in Wallaroo with a low-overhead."
+tags = [
+    "metrics",
+    "performance",
+    "wallaroo"
+]
+categories = [
+    "Metrics",
+    "Performance",
+]
++++
+
+Metrics play an integral part in providing confidence in a high-performance software system. Whether you’re dealing with a data processing framework or a web server, metrics provide insight into whether your system is performing as expected. A direct impact of capturing metrics is the performance cost to the system they're for.
+
+When we first started development on [Wallaroo](https://github.com/WallarooLabs/Wallaroo), our high-throughput, low-latency, and elastic data processing framework, we were aiming to meet growing demand in the data processing landscape. Companies are relying on more and more data all while expecting to process that data as quickly as possible for, among other things, faster, better decision making, and cost reduction. Because of this, we set out to develop Wallaroo with three core principles in mind:
+
+1. High-Throughput
+2. Low-Latency
+3. 1/3 of the infrastructure costs/needs compared to existing solutions.
+
+Proving to our users that we were delivering on these principles was an essential part of the early development cycle of Wallaroo. How could we prove we were delivering on these principles? We needed to capture metrics. Metrics are an excellent way to spot bottlenecks in your system. Providing meaningful metrics to our users would mean they could make quicker iterations in their development cycle by being able to spot specific bottlenecks in various locations of our system. An added benefit that came from capturing these metrics was that it also sped up the Wallaroo development cycle as we were able to use them as guidelines to [measure the “cost” of every feature](https://blog.wallaroolabs.com/2017/06/whats-the-secret-sauce/#measure-the-cost-of-every-feature).
```

Comment:
`Metrics are an excellent way to spot bottlenecks in your system.` seems out of place because the rest of the paragraph is talking about `our system` and `our users`. Maybe change it to be `Metrics are an excellent way to spot bottlenecks in any system.`?